terra/compute/container.py

```diff
@@ -153,3 +153,46 @@ def add_volume(self, local, remote, flags=None, prefix=None,
 
     self.volumes.append((local, remote))
     self.volumes_flags.append(flags)
+
+  def add_volume_input(self, local, remote):
+    '''
+    Add a read-only input volume, confirming if the local file/folder
+    exists. Raise a :obj:`ValueError` should validation fail.
+    '''
+
+    self.add_volume(local, remote, flags='ro',
+                    local_must_exist=True)
+
+  def add_file_input(self, local, remote, use_local_extension=False):
+    '''
+    Add a read-only input file, confirming that the local file exists.
+    Maintain local file extension if requested.
+    '''
+
+    # replace remote extension with local extension
+    if use_local_extension:
+      local_ext = os.path.splitext(local)[1].lower()
+      remote = os.path.splitext(remote)[0] + local_ext
+
+    # update volume
+    self.add_volume_input(local, remote, local_must_exist=True)
```

Comment:
I found on at least one project that calls certain files/directories "inputs" this can get confusing.  Should these be renamed to `add_volume_readonly` and `add_file_readonly`?